packages/trackerless-network/src/logic/inspect/InspectSession.ts

```diff
@@ -0,0 +1,49 @@
+import { EventEmitter } from 'eventemitter3'
+import { MessageRef } from '../../proto/packages/trackerless-network/protos/NetworkRpc'
+import { PeerIDKey } from '@streamr/dht'
+
+export interface Events {
+    done: () => void
+}
+
+interface InspectSessionConfig {
+    inspectedPeer: PeerIDKey
+}
+
+const createMessageKey = (messageRef: MessageRef): string => {
+    return `${messageRef.messageChainId}:${messageRef.timestamp}:${messageRef.sequenceNumber}`
+}
+export class InspectSession extends EventEmitter<Events> {
+    
+    // Boolean indicates if the message has been received by the inspected node
+    private readonly inspectionMessages: Map<string, boolean> = new Map()
```

Comment:
Could use set of strings? If there is a string, it means the message has been received by the inspected node

Reply:
We need to keep track of if a message have been received from the inspected peer and any of the stream neighbors. The session is done after any message has been seen from any of the inspectors stream neighbors and the inspected peer.